moc_openshift.py

```diff
@@ -469,3 +541,108 @@ def update_rolebindings(self, project_name, role, rolebindings_json):
+            if scope not in quota_def:
+                quota_def[scope] = {}
+            quota_def[scope][quota_name] = quota_spec[mangled_quota_name]
+        # create the openshift quota resources
+        quota_create_status_code = 200
+        quota_create_msg = ""
+        for scope, quota_item in quota_def.items():
+            resource_quota_json = {
+                "apiVersion": "v1",
+                "kind": "ResourceQuota",
+                "metadata": {"name": f"{project_name.lower()}-{scope.lower()}"},
+                "spec": {"hard": {}},
+            }
+            if scope != "Project":
+                resource_quota_json["spec"]["scopes"] = []
+                resource_quota_json["spec"]["scopes"].append(scope)
+            non_null_quota_count = 0
+            for quota_name in quota_item:
+                if quota_item[quota_name] is not None:
+                    non_null_quota_count += 1
+                    resource_quota_json["spec"]["hard"][quota_name] = quota_item[
+                        quota_name
+                    ]["value"]
+            if non_null_quota_count > 0:
+                url = (
+                    f"{self.get_url()}/api/v1/namespaces/{project_name}/resourcequotas"
+                )
+                resp = self.post_request(url, resource_quota_json, True)
+                time.sleep(2)
+                if resp.status_code in [200, 201]:
+                    quota_create_msg = f"{quota_create_msg} Quota {project_name}/{quota_name} successfully created\n"
+                else:
+                    if resp.status_code > quota_create_status_code:
+                        quota_create_status_code = resp.status_code
+                quota_create_msg = f"{quota_create_msg} Quota {project_name}/{quota_name} creation failed"
+            if quota_create_status_code == 200:
+                quota_create_msg = f"All quota from {project_name} successfully created"
+        return Response(
+            response=quota_create_msg,
+            status=quota_create_status_code,
+            mimetype="application/json",
+        )
+
+    def get_resourcequotas(self, project_name) -> list:
+        """Returns a dictionary of all of the resourcequota objects"""
+        url = f"{self.get_url()}/api/v1/namespaces/{project_name}/resourcequotas"
+        rq_data = self.get_request(url, True).json()
+        self.logger.info(pprint.pformat(rq_data))
+        rq_list = []
+        for rq_name in rq_data["items"]:
+            rq_list.append(rq_name["metadata"]["name"])
+        return rq_list
+
+    def delete_quota(self, project_name, resourcequota_name):
+        """In an openshift namespace {project_name) delete a specified resourcequota"""
+        url = f"{self.get_url()}/api/v1/namespaces/{project_name}/resourcequotas/{resourcequota_name}"
+        payload = {
+            "kind": "DeleteOptions",
+            "apiVersion": "user.openshift.io/v1",
+            "gracePeriodSeconds": 300,
```

Comment:
Why are we setting `gracePeriodSeconds` here? According to the docs, this is:

> The duration in seconds before the object should be deleted. Value must be non-negative integer. The value zero indicates delete immediately. If this value is nil, the default grace period for the specified type will be used. Defaults to a per object value if not specified. zero means delete immediately.

Don't we want to delete things immediately?

Reply:
Since you are asking, I'll have to test this again.

As I remember, in openshift 3.x, I couldn't delete certain objects because I didn't specify a gracePeriodSeconds field, got a different error when I made it 0, so found an example that worked and went with it.  It never took 5 minutes, however, that worked for every object that I was deleting, so I just went with it.

I was also under the impression that if you specified a grace period, that it would gracefully bring things down.

Reply:
ok, have eliminated the payload for the delete identity.  This was required on openshift 3.x, but doesn't seem to be required now.

Reply:
This is no longer needed and has been removed.  It makes things a bit easier in 4.x.